thoth/user_api/api_v1.py

```diff
@@ -252,14 +263,18 @@ def post_provenance_python(
     kebechet_metadata: typing.Optional[typing.Dict[str, typing.Any]] = None,
 ):
     """Check provenance for the given application stack."""
-    protected_fields = ["kebechet_metadata"]
     parameters = locals()
+    token = parameters.pop("token", None)
 
-    if Configuration.API_TOKEN == token:
-        for k in protected_fields:
+    authenticated = False
+    if token is not None:
+        if Configuration.API_TOKEN != token:
+            return {"error": "Bad token supplied"}, 401
+
+        authenticated = True
+        for k in _PROVENANCE_CHECK_PROTECTED_FIELDS:
             if parameters[k] is not None:
-                response = {"error": "Attempted to set protected field without proper permissions."}
-                return response, 401
+                return {"error": f"Parameter {k!r} requires token to be set to perform authenticated request"}, 401
```

Comment:
sorry I'm confused by the conditions: 
if token is set and token is configuration.api_token, what is a case where we hit the last return (return {"error": f"Parameter {k!r} requires token to be set to perform authenticated request"}, 401)?

Reply:
Now I understand :) Thanks @fridex!!